### ISSUE TITLE
[User Management] Create reviewer role

### ISSUE THREAD

**Issue by jh-RLI** [OpenEnergyPlatform/oeplatform#1200]
As #1149 indicates there is the need for a reviewer role for the Open Peer Review. 

- [ ] Extend the login model
- [ ] Setup access-permissions

**Comment by jh-RLI** [comment]
I noticed that we haven't quite decided how to handle permissions related to the review. I once suggested introducing a role for the reviewer, as suggested in this issue. But we have also decided that anyone can do a review. In this case it would not be necessary to introduce a role. Still it is necessary to be able to identify the contributor and the reviewer who are involved somehow. We have also decided to introduce a moderator role that can access all available reviews and perform some moderator actions or close reviews if they are outdated, for example.

We have to decide:
- Do we really want to give every user the ability to rate? (This needs to be decided soon in order to proceed with parts of the development).
- What are the necessary actions of the moderator? (can be done later)

@christian-rli @Ludee @Darynarli

**Comment by jh-RLI** [comment]
I have found a feasible way and have started the implementation.

The review itself will be stored in a special table in dataedit. This table will contain new fields containing the user ID of the author (at the moment probably the table author ... later maybe all users included in a permission group) and also the user ID of the reviewer. I think this will make it much easier to link the entities. 

To make the reviews visible to the reviewers, I would suggest creating a user group. All users assigned to this group can start a review when they see a table in model_draft that has not been reviewed before. The group design also allows us to manage the reviewers in case a user does not follow the guidelines.  Nonetheless, this group can be assigned to any user initially. 

I also propose to somehow limit the users who qualify as reviewers in the future. We could introduce a course that has to be taken in order to qualify as a reviewer. This would help to increase the quality of the open peer review process.

**Comment by Darynarli** [comment]
1. you can give review access to users who have already been logged in for a certain time and have shown some activity and contributed to oep. And also to those who have passed some kind of test or course. Later it will be possible to do as I described earlier.
2. those who control reviewers, are they moderators?
3. recommendations can be written somewhere

**Comment by jh-RLI** [comment]
1. and 4. Yes, I had a similar idea. I think at first we will just allow every user to do a review and then when the review feature in the oep is working properly, we will work on the restrictions to become a reviewer. I don't think anything bad will happen immediately if in the beginning all users have permission to write a review.
2. Yes the moderator will be introduced to moderate all user content on the oep but initially just the reviews. They will be able to delete stale reviews and they will also be able to remove users from the reviewer group.
3. What recommendations?

**Comment by Darynarli** [comment]
> 3. What recommendations?
Recommendations for the reviewer on how to correctly evaluate and in what time frame, and for the user, in what time frame you need to give an answer and how to post the article correctly.

**Comment by jh-RLI** [comment]
Good idea. We have to create tooltips and some learning material in the academy.

**Comment by christian-rli** [comment]
This comment comes late, I'm aware, but I approve of your approach :) Improvements can be implemented when needed.
